### DISCUSSION TITLE
Multiple clients do not appears as "multiple mouse"

### DISCUSSION THREAD

**Discussion by shodanx2** [LizardByte/community#494] [Sunshine]
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Is your issue described in the documentation?

- [x] I have read the documentation

### Is your issue present in the latest beta/pre-release?

yes

### Describe the Bug

Hello,
Me and my friends share a single computer with sunshine/moonlight
We each have moonlight running on second monitor in absolute mouse mode (no mouse capture)
Works fantastic, we're really able to all use the same computer at the same time, it's very pro-social.

Issue I'm having is I would like each user to have their own cursors.
So I installed eithermouse (.com) (it is freeware, another option would have been paid software mousemux, as far as I can tell, there is no open source software for this application)

Here is an sample of what it should have looked like

https://github.com/user-attachments/assets/936be856-a6bc-4f5a-8349-0d0bbf64eef9

Unfortunately, it appears there is a problem with how sunshine emulated mouse events from multiple clients, they all go through a single instance of the virtual HID device, hence multiple mouse labelling software cannot distinguish between multiple cursors.

### Expected Behavior

Each client should use a separate HID instance instead of multiplexing all of them together

### Host Operating System

windows

### Operating System Version

Windows 10 22h2 19045.2006

### Architecture

64 bit

### Sunshine commit or version

0.23.1

### Package

Linux - AppImage

### GPU Type

nvidia

### GPU Model

gtx 1060

### GPU Driver/Mesa Version

31.0.15.3699

### Capture Method

None

### Config

```Shell
channels = 4
```

### Apps

desktop

### Relevant log output

```shell
Log was too big to paste 
https://pastebin.com/aE8QjyTP
However I don't think any line of the log is actually relevant to this issue
```

**Comment by ReenigneArcher** [comment]
Sunshine is working as intended, I've converted this to a feature request.

**Reply by shodanx2** [reply]
thanks !
Do you have the link please ?

**Comment by ReenigneArcher** [comment]
As far as I know, this isn't possible.

Connect two physical mice to your computer and see what happens.

**Reply by shodanx2** [reply]
Hi, see the video screenshot in the first post as demonstration that it is in fact possible.

**Reply by ReenigneArcher** [reply]
Hmm... we do create HID mice now if using the new Virtual HID Driver, but I don't think we're creating a second HID mouse for each client. That would be rather easy; however there's no mechanisms in Sunshine control separate mice.

This is a pretty niche use case I think. PRs welcome I guess.

**Reply by shodanx2** [reply]
Yep, just needs to create a separate HID for each client.
I don't know about being "niche" I think it's very common to screen share with friends.
But yes, changing sunshine to use a virtual driver was the hard part. 
I will probably do it once I have the time

Here is the actual issue regarding this topic

https://github.com/LizardByte/Sunshine/issues/3044

**Reply by ReenigneArcher** [reply]
It's niche, this has been open for over 2 years and doesn't have any upvotes.

It's a little more complicated than just creating separate HID devices for each client. We also need to keep track of which client is controlling which mouse, and probably refactor a ton of code actually.
